testsuite/summarize_tests.py

```diff
@@ -97,10 +102,13 @@ def parse_result_file(fname):
     print(tline)
     for pn, pr in results.items():
         print(f"{pn:<{first_col_w}s}", end="")
-        for c in cols:
-            fmt = ".1f" if c == "Time" else "d"
-            print(f"{pr[c]:{col_w}{fmt}}", end="")
-        print()
+        if pr["Tests"] == 0 and pr["Failed tests"]:
+            print(f"{'--- XML PARSING FAILURE ---':^{len(cols) * col_w}}")
+        else:
```

Comment:
Not necessary if marked as in previous comment. The extra `XML PARSING FAILURE` string can be absorbed into `msg` and I don't see why centering necessary in addition to all-caps.

Reply:
I have removed the all caps. As explained above, we need to treat this case special. And I find the centred message much more visually pleasing :).